### PR TITLE
added possibility to use MkdirLockFile as an alternative

### PR DESCRIPTION
**Pull request by psychok7**
I decided to create this PR because of https://github.com/pinax/django-mailer/issues/43 and in the end i realized that the problem was that i was using my `Vagrant` machine for development and it was not locking using the default `FileLock` from lockfile in my shared folder between the Guest and Host (maybe its related to https://github.com/mitchellh/vagrant/issues/713 ). 

in the `aquire()` function in the os.link() it was giving me `OSError: [Errno 1] Operation not permitted` 

I then realized that lockfile also allows `MkdirLockFile` that provides a `mkdir(2)() system call as the basic lock mechanism` and it is even used on Windows systems.

Basically this PR allows the user to choose if he wants to use `FileLock` or `MkdirLockFile` where `FileLock` continues being the default.

### CODE REVIEW

**Comment by paltman** [comment]
@spookylukey we have a Slack setup for Pinax now with a channel dedicated to `django-mailer`. Want to join?  http://slack.pinaxproject.com

**Comment by psychok7** [comment]
@spookylukey does this seem reasonable to you??

**Comment by spookylukey** [comment]
Actually, it seems overkill to have a setting for this.

If you find you need to change the setting, you've already wasted a lot of time debugging, probably. It should Just Work all the time, or have an explicit error if it can't work.

**Comment by ossanna16** [comment]
Any further discussion on this? Can this PR be closed?
